src/selectors/ProfilePageSelector.js

```diff
@@ -0,0 +1,119 @@
+import { createSelector } from 'reselect';
+
+export const formIdSelector = (state, props) => props.formId;
+export const authenticationUsernameSelector = state => state.authentication.username;
+export const profileAccountSelector = state => state.profilePage.account;
+export const profileCertificatesSelector = state => state.profilePage.certificates;
+export const profileAccountDraftsSelector = state => state.profilePage.accountDrafts;
+export const profileVisibilityDraftsSelector = state => state.profilePage.visibilityDrafts;
+export const profileVisibilitySelector = state => state.profilePage.preferences.visibility;
+export const saveStateSelector = state => state.profilePage.saveState;
+export const currentlyEditingFieldSelector = state => state.profilePage.currentlyEditingField;
+export const accountErrorsSelector = state => state.profilePage.errors;
+
+export const isCurrentUserProfileSelector = createSelector(
+  authenticationUsernameSelector,
+  profileAccountSelector,
+  (username, profileAccount) => username === profileAccount.username,
+);
+
+export const editableFormModeSelector = createSelector(
+  profileAccountSelector,
+  profileCertificatesSelector,
+  formIdSelector,
+  isCurrentUserProfileSelector,
+  currentlyEditingFieldSelector,
+  (account, certificates, formId, isCurrentUserProfile, currentlyEditingField) => {
```

Comment:
FWIW I'm not a huge fan of this composable selector pattern

Reply:
It's a bit weird to get used to, admittedly.  It really starts shining when you need lots of different kinds of data in lots of different areas of the app...